### PR TITLE
feat(aws): S3 storage lens configurations

### PR DESCRIPTION
**Pull request by disq**
Implements [#1383](https://github.com/cloudquery/cloudquery/issues/1383)

The S3 export configuration can be found under `aws_s3_lens_configurations.data_export->'S3BucketDestination'`... I don't think we should read the files and import the actual metrics? @bbernays 

**UPDATE**
I can't seem to get this return any data, even tried with `aws_debug` but `aws s3control list-storage-lens-configurations --account-id=XXXX` works for me with the same credentials. Tested the region thing etc. using the aws cli as well.

Related: https://github.com/aws/aws-sdk/issues/259
`aws_debug` wasn't enough so had to add `aws.LogResponseWithBody` to the client log mode, turns out the issue above matches our findings.

### CODE REVIEW

File: plugins/source/aws/codegen/recipes/s3.go [hunk 1]
```diff
@@ -107,6 +108,27 @@ func S3Resources() []*Resource {
 					},
 				}...),
 		},
+		{
+			SubService:          "lens_configurations",
```
Comment (disq) [inline plugins/source/aws/codegen/recipes/s3.go:112]: `s3_storage_lens_configurations`? I think that's too long so I went for "lens*"

**Review by erezrokah — APPROVED** [review]

File: plugins/source/aws/codegen/recipes/s3.go (hunk 1 above)
Comment (hermanschaaf) [reply plugins/source/aws/codegen/recipes/s3.go:112]: `s3_storage_lens_configurations` seems closer to the underlying SDK - AWS calls it `StorageLensConfigurations` so I think we should keep it as similar as possible, as long as we don't exceed table name length limits (which for `s3_storage_lens_configurations` we don't).

**Review by hermanschaaf — CHANGES_REQUESTED** [review]
LGTM, let's just decide what the name should be

**Comment by disq** [comment]
Also there's a 'Home Region' concept, I'm not sure if it would have an effect on our requests. Looks like it shouldn't, but I'll test it in a bit.

**Comment by disq** [comment]
I can't seem to get this return any data, even tried with `aws_debug` (I don't see any data in the response either) but  `aws s3control list-storage-lens-configurations --account-id=XXXX` works.

Could be related: https://github.com/aws/aws-sdk/issues/259

**Comment by erezrokah** [comment]
What is the next step for this PR? Contacting AWS support?

**Comment by Hokwang** [comment]
@disq Hi, I didn't read all text but did you check https://github.com/aws/aws-sdk-go-v2 ?
I think https://github.com/aws/aws-sdk is not managed these days.

**Pull request closed without merging**